src/sugar/autofill.ts

```diff
@@ -193,4 +200,19 @@ async function setLatestValidatedLedgerSequence(
   tx.LastLedgerSequence = ledgerSequence + LEDGER_OFFSET
 }
 
+async function checkAccountDeleteBlockers(client: Client, tx: AccountDelete) {
+  const request: AccountObjectsRequest = {
+    command: 'account_objects',
+    account: tx.Account,
+    ledger_index: 'validated',
+    deletion_blockers_only: true,
+  }
+  const response = await client.request(request)
+  if (response.result.account_objects.length > 0) {
+    throw new XrplError(
+      `Account ${tx.Account} cannot be deleted; there are Escrows, PayChannels, RippleStates, or Checks associated with the account.`,
```

Comment:
Could we also include the verbatim `account_objects` response so clients can see what the deletion blockers are?